concrete/src/Package/Package.php

```diff
@@ -367,205 +580,279 @@ public function uninstall()
+            // Step 4 - Check for package dependencies
+            $packageService = $this->app->make(PackageService::class);
+            /* @var PackageService $packageService */
+            $installedPackageHandles = $packageService->getInstalledHandles();
+            $myDependencies = $this->getPackageDependencies();
+            foreach ($myDependencies as $packageHandle => $requirements) {
+                if (in_array($packageHandle, $installedPackageHandles)) {
+                    if ($requirements === false) {
+                        $errors[] = t('This package can\'t be installed if the package with handle %s is already installed', $packageHandle);
+                    } elseif ($requirements !== true) {
+                        $packageVersion = $packageService->getByHandle($packageHandle)->getPackageVersion();
+                        if (is_array($requirements)) {
+                            if (version_compare($packageVersion, $requirements[0]) < 0 || version_compare($packageVersion, $requirements[1]) > 0) {
+                                $errors[] = t('This package requires the package with handle %1$s has a version from %2%s to %3$s', $packageHandle, $requirements[0], $requirements[1]);
+                            }
+                        } else {
+                            if (version_compare($packageVersion, $requirements) < 0) {
+                                $errors[] = t('This package requires the package with handle %1$s has a version %2%s or greater', $packageHandle, $requirements);
+                            }
+                        }
+                    }
+                } else {
+                    if ($requirements === true) {
+                        $errors[] = t('Before installing this package, you need to install the package with handle %s', $packageHandle);
+                    } elseif (is_array($requirements)) {
+                        $errors[] = t('Before installing this package, you need to install the package with handle %1s (version between %2$s and %3$s)', $packageHandle, $requirements[0], $requirements[1]);
+                    } elseif ($requirements !== false) {
+                        $errors[] = t('Before installing this package, you need to install the package with handle %1s, version %2$s or greater', $packageHandle, $requirements);
+                    }
+                }
             }
+            $myPackageHandle = $this->getPackageHandle();
+            foreach ($installedPackageHandles as $packageHandle) {
+                $packageController = $packageService->getClass($packageHandle);
+                $packageDependencies = $packageController->getPackageDependencies();
+                if (isset($packageDependencies[$myPackageHandle]) && $packageDependencies[$myPackageHandle] === false) {
+                    $errors[] = t('This package can\'t be installed if the package with handle %s is already installed', $packageHandle);
```

Comment:
`This package can\'t be installed if the package with handle %s is already installed` -> s/if/because/ ?

Reply:
> `This package can\'t be installed if the package with handle %s is already installed` -> s/if/because/ ?

... I don't understand...

Reply:
If you say `if` it's a condition. But I think you're giving a reason here, so `because` would be appriopriate.

Reply:
:+1: ⇒ https://github.com/concrete5/concrete5/pull/5995/commits/1e1ae7d129e584961e00a98158d1f4976eb2a652